### PR TITLE
Use loguru

### LINKED ISSUE #301: Using loguru
I have been using loguru for a year now and I really like it. I think this could make the logging much easier for us.

There is the user aspect. 
One could just do `logger.add("alchemlyb.log")` to save the log to a file.

Also one doesn't need to initialise a logger and the function name would just be printed to the start of the line.
For debugging purpose, the loguru would also print all the public variables to the log.

### PR DESCRIPTION
**Pull request by xiki-tempula**
Switch to use loguru for logging
Fix #301

### CODE REVIEW

**Comment by xiki-tempula** [comment]
@orbeckst Do you mind have a check to see if you are happy with the new logger? How it works with other section of your code. I'm working on the doc now, where I plan to add a new miscellaneous section to the top level doc and a second level logging. Then in other places, I will just refer to this section.

File: docs/miscellaneous/logging.rst [hunk 1]
```diff
@@ -0,0 +1,29 @@
+.. _logging_section:
+
+Logging
+=======
+
+In **alchemlyb**, we use :mod:`loguru` for logging. By default, the
```
Comment (orbeckst) [inline docs/miscellaneous/logging.rst:6]: add the appropriate intersphinx links to conf.py to make links to loguru docs work

File: docs/miscellaneous/logging.rst [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+:mod:`loguru` will print the logging information into the `stderr`.
```
Comment (orbeckst) [inline docs/miscellaneous/logging.rst:7]: Can you link to `sys.stderr` in the Python docs?

File: docs/miscellaneous/logging.rst [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+Print to the stderr
+-------------------
+
+If you want to customise the printing to the `stderr`, you could remove the
+existing sink first ::
+
+    from loguru import logger
+    logger.remove()
+
+Then add other custom sink ::
```
Comment (orbeckst) [inline docs/miscellaneous/logging.rst:18]: Link to a page that explains what the format does, filter etc — basically the docs for `logger.add`

File: docs/miscellaneous/logging.rst [hunk 4]
```diff
@@ -0,0 +1,29 @@
+.. _logging_section:
+
+Logging
+=======
+
+In **alchemlyb**, we use :mod:`loguru` for logging. By default, the
+:mod:`loguru` will print the logging information into the `stderr`.
+
```
Comment (orbeckst) [inline docs/miscellaneous/logging.rst:8]: Add a link to https://loguru.readthedocs.io/en/stable/overview.html#entirely-compatible-with-standard-logging along the lines of
```reST
The loguru logger is compatible with the :mod:`logging` module of the Python
standard library and can easily be `configured to log to a *logging* handler<logguru-standard-logging>`_. 

.. _logguru-standard-logging:
    https://loguru.readthedocs.io/en/stable/overview.html#entirely-compatible-with-standard-logging
```

File: src/alchemlyb/estimators/mbar_.py [hunk 5]
```diff
@@ -25,12 +23,12 @@ class MBAR(BaseEstimator, _EstimatorMixOut):
...
     method : str, optional, default="robust"
         The optimization routine to use.  This can be any of the methods
-        available via :func:`scipy.optimize.minimize` or 
+        available via :func:`scipy.optimize.minimize` or
         :func:`scipy.optimize.root`.
 
     verbose : bool, optional
         Set to ``True`` if verbose debug output from :mod:`pymbar` is desired.
-        Output from alchemlyb is logged via :mod:`logging`.
+        Output from alchemlyb is logged via :mod:`loguru`.
```
Comment (orbeckst) [inline src/alchemlyb/estimators/mbar_.py:31]: remove as no logging done here?

File: docs/miscellaneous.rst [hunk 6]
```diff
@@ -0,0 +1,11 @@
+Miscellaneous
+=============
+
+This page includes aspects that would improve your usage of **alchemlyb**.
+
+.. toctree::
+   :maxdepth: 2
+   :caption: Miscellaneous
```
Comment (orbeckst) [inline docs/miscellaneous.rst:8]: I'd remove the TOC caption — looks superfluous as it is repeated, see https://alchemlyb--303.org.readthedocs.build/en/303/miscellaneous.html

**Review by orbeckst — CHANGES_REQUESTED** [review]
minor comments, overall good

File: docs/miscellaneous.rst (hunk 6 above)
Comment (xiki-tempula) [reply docs/miscellaneous.rst:8]: Sorry, what do you mean by this? I could change the` :maxdepth: 2` to 1, if Print to the stderr and Save to a file is superfluous?

Comment (xiki-tempula) [reply docs/miscellaneous.rst:8]: Oh, I see.

**Comment by xiki-tempula** [comment]
@orbeckst Thanks for the review. I have addressed the changes.

**Pull request merged**